lib/registration.pm

```diff
@@ -119,12 +135,20 @@ sub fill_in_registration_data {
                     save_screenshot;
                 }
             }
-            send_key 'alt-n';    # next
-            while (check_screen('import-untrusted-gpg-key', 20)) {
-                record_soft_failure 'untrusted gpg key';
-                send_key 'alt-t', 2;
+            send_key 'alt-n';
+            # start addons/modules registration, it needs longer time if select multiple or all addons/modules
+            while (assert_screen(['import-untrusted-gpg-key', 'yast_scc-pkgtoinstall', 'inst-addon'], 120)) {
```

Comment:
this is analogue to while(1) { assert_screen(qw(tag1 tag2)); if match_has_tag('tag2') last; }

and I consider the while(assert_screen(qw(tag1 tag2)) { if match_has_tag('tag2') last; } easier to read and more to the point.

Yes, the return value doesn't matter, but it will loop and that's what matters.